packages/oscal-react-library/src/components/OSCALCatalogGroups.tsx

```diff
@@ -68,23 +75,31 @@ TabPanel.propTypes = {
  * Validates if the lowest control provided in an decending list (controlLayers) can be found
  * within the provided groups.
  *
- * @param {*} groups The control groupings
- * @param {*} controlLayers An decending list of groups/controls
- * @param {*} rootLayer The top most layer
- * @returns This returns the lowest found control or null if not
+ * @param groups The control groupings
+ * @param controlLayers An decending list of groups/controls
+ * @param rootLayer The top most layer
+ * @returns This returns the lowest found control or undefined if not
  */
-function determineControlExists(groups, controlLayers, rootLayer) {
+function determineControlExists(
+  groups: ControlGroup[] | undefined,
+  controlLayers: string[],
+  rootLayer: string
+): ControlGroup | undefined {
   // Ensure catalog tab grouping exists
   let upperLayer = groups?.find(
     (group) => group?.id === rootLayer || conformLinkIdText(group?.title) === rootLayer
   );
+
   if (!upperLayer) {
-    return null;
+    return undefined;
   }
+
   // Ensure lowest/deepest control exists within controls/groups
   for (let i = 1; i < controlLayers.length && upperLayer; i += 1) {
-    const subControl = upperLayer?.controls?.find((control) => control.id === controlLayers[i]);
-    const subGroup = upperLayer?.groups?.find(
+    const subControl: Control | undefined = upperLayer?.controls?.find(
+      (control) => control.id === controlLayers[i]
+    );
+    const subGroup: ControlGroup | undefined = upperLayer?.groups?.find(
```

Comment:
I am confused about why this needs a specific type -- what part of this causes `tsc` to lose the ability to infer types?

Reply:
```
'subControl' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer. [7022]

'subGroup' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer. [7022]
```